crates/biome_css_semantic/src/semantic_model/model.rs

```diff
@@ -188,21 +240,46 @@ impl std::fmt::Display for Specificity {
 /// ```
 #[derive(Debug, Clone)]
 pub struct CssDeclaration {
-    pub property: CssProperty,
-    pub value: CssValue,
-    pub range: TextRange,
+    pub(crate) node: CssSyntaxNode,
+    pub(crate) property: CssProperty,
+    pub(crate) value: CssValue,
 }
 
-#[derive(Debug, Clone, Default)]
-pub struct CssProperty {
-    pub name: String,
-    pub range: TextRange,
+impl CssDeclaration {
+    pub fn node(&self) -> &CssSyntaxNode {
+        &self.node
+    }
+
+    pub fn range(&self) -> TextRange {
+        self.node.text_trimmed_range()
+    }
+
+    pub fn property(&self) -> &CssProperty {
+        &self.property
+    }
+
+    pub fn value(&self) -> &CssValue {
+        &self.value
+    }
+}
+
+declare_node_union! {
+    pub CssProperty = CssDashedIdentifier | CssIdentifier
+}
+
+impl CssProperty {
+    pub fn value(&self) -> SyntaxResult<TokenText> {
+        let token = match self {
+            CssProperty::CssDashedIdentifier(node) => node.value_token()?,
+            CssProperty::CssIdentifier(node) => node.value_token()?,
+        };
+
+        Ok(token.token_text_trimmed())
+    }
 }
 
-#[derive(Debug, Clone, Default)]
-pub struct CssValue {
-    pub text: String,
-    pub range: TextRange,
+declare_node_union! {
+    pub CssValue = CssGenericComponentValueList | CssComposesPropertyValue
```

Comment:
I tried to add some methods, but I couldn't because the `ComposePropertyValue` syntax from the CSS module is quite unique:( I think we can leave it as it is for now